tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -244,3 +246,123 @@ def test_all_transactions(self):
+
+    @cluster(num_nodes=3)
+    def test_delete_topic_from_tx(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        txs_info = self.admin.get_all_transactions()
+        assert len(
+            txs_info) == 1, "Should be only one transaction in current time"
+
+        rpk = RpkTool(self.redpanda)
+        topic_name = self.topics[0].name
+        rpk.delete_topic(topic_name)
+
+        tx = txs_info[0]
+        assert tx[
+            "transactional_id"] == tx_id, f"Expected transactional_id: {tx_id}, but got {tx['transactional_id']}"
+
+        for partition in tx["partitions"]:
+            assert (partition["ns"] == "kafka")
+            if partition["topic"] == topic_name:
+                partition_info = {
+                    "namespace": partition["ns"],
+                    "topic": partition["topic"],
+                    "partition_id": partition["partition_id"],
+                    "etag": partition["etag"]
+                }
+                self.admin.delete_partition_from_transaction(
+                    tx["transactional_id"], partition_info)
+
+        producer.commit_transaction()
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            if topic.name is not topic_name:
+                for partition in range(topic.partition_count):
+                    producer.produce(topic.name, '0', '0', partition)
+
+        producer.commit_transaction()
+
+    @cluster(num_nodes=3)
+    def test_delete_topic_from_tx_check_error(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        error_topic_name = "error_topic"
+
+        for topic in self.topics:
+            assert error_topic_name is not topic.name
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        partition_info = {
+            "namespace": "kafka",
+            "topic": error_topic_name,
+            "partition_id": 0,
+            "etag": 0
+        }
+        try:
+            self.admin.delete_partition_from_transaction(tx_id, partition_info)
+        except requests.exceptions.HTTPError as e:
+            assert e.response.status_code == 400
```

Comment:
can you either (1) add a comment here about what is happening or (2) assert on something more specific so the check is obvious. I *think* this is checking that an error occurs when the partition_info["topic"] doesn't exist, but it isn't clear. fyi, you can also use `e.response.text` as a way to assert something more specific like "topic not found" in e.response.text etc...

EDIT: this seems to be the only place `error_topic_name = "error_topic"` is referenced. if this is testing something about the `delete_partition_from_transaction` interface behavior for a non-existent topic, then should it be a separate test?

Reply:
I added this test to check errors.